### PR TITLE
SNOW-976703 : Adding cummulative aggregation dataframe function

### PR DESCRIPTION
**Pull request by sfc-gh-rsureshbabu**
Please answer these questions before submitting your pull requests. Thanks!

1. What GitHub issue is this PR addressing? Make sure that there is an accompanying issue to your PR.

SNOW-976703

The Pr is adding the cummulative aggregation function as proposed in https://docs.google.com/document/d/14J5lr_a3fE1xeU-YzKWdeduBdC_035sry11lJUjjkT8/edit

2. Fill out the following pre-review checklist:

   - [x] I am adding a new automated test(s) to verify correctness of my new code
   - [ ] I am adding new logging messages
   - [ ] I am adding a new telemetry message
   - [ ] I am adding new credentials
   - [ ] I am adding a new dependency

3. Please describe how your code solves the related issue.

   Please write a short description of how your code change solves the related issue.

### CODE REVIEW

File: src/snowflake/snowpark/dataframe_transform_functions.py [hunk 1]
```diff
@@ -120,3 +120,67 @@ def moving_agg(
...
+    ) -> "snowflake.snowpark.dataframe.DataFrame":
+        """
+        Applies cummulative aggregations to the specified columns of the DataFrame using defined window direction,
+        and grouping and ordering criteria.
+
+        Args:
+            aggs: A dictionary where keys are column names and values are lists of the desired aggregation functions.
+            order_by: A list of column names that specify the order in which rows are processed.
+            group_by: A list of column names on which the DataFrame is partitioned for separate window calculations.
+            direction: A string indicating the direction of accumulation ('forward' or 'backward').
```
Comment (sfc-gh-sfan) [inline src/snowflake/snowpark/dataframe_transform_functions.py:140]: Just curious if this should be a boolean, no strong opinion here.

**Review by sfc-gh-sfan — COMMENTED** [review]
I know this is a PR on top of the base PR that introduced `dataframe_transform_functions`, but let's make sure we also add this new function to `CHANGELOG`.

File: src/snowflake/snowpark/dataframe_analytics_functions.py [hunk 2]
```diff
@@ -168,3 +168,83 @@ def moving_agg(
...
+            aggs: A dictionary where keys are column names and values are lists of the desired aggregation functions.
+            order_by: A list of column names that specify the order in which rows are processed.
+            group_by: A list of column names on which the DataFrame is partitioned for separate window calculations.
+            is_forward: A boolean indicating the direction of accumulation. True for 'forward' and False for 'backward'.
+            col_formatter: An optional function for formatting output column names, defaulting to the format '<input_col>_<agg>'.
+                        This function takes two arguments: 'input_col' (str) for the column name, 'operation' (str) for the applied operation,
+                        and returns a formatted string for the column name.
+
+        Returns:
+            A Snowflake DataFrame with additional columns corresponding to each specified cumulative aggregation.
```
Comment (sfc-gh-yixie) [inline src/snowflake/snowpark/dataframe_analytics_functions.py:194]: Nit: Snowpark DataFrame

**Review by sfc-gh-yixie — APPROVED** [review]
LGTM with two small doc comments.

**Pull request merged**